### ISSUE TITLE
Finetune DINO V3 on Uchen Ume class

### ISSUE THREAD

**Issue by karmatai** [OpenPecha/BEC#302]
# Description
In this task we are going to group all the classes into two Parent classes, Uchen and Ume and then finetune DINO v3 on these class and save the weight for later use. 

**Repo:** https://github.com/OpenPecha/script-classification-model-train/tree/main
**Model:** https://huggingface.co/karma689/uchen-ume-classifier

## Subtasks
- [x] Set up Vast AI ssh, download data and clone the script.
- [x] Group the data into two parent classes.
- [x] Implement preprocessing steps
- [x] Build script and Finetune DINO V3 on two grouped class.
- [x] Evaluate and write result.
- [x] Update hugging face.
- [x] Rerun the training script.
- [x] Save the dataset and model checkpoint.
- [x] Select benchmark dataset and ask for review from thinlay.
- [x] Convert to parquet file.
- [x] Upload to the hugging face.

## Result
Metric | Stage A (Best: Ep 6) | Stage B (Best: Ep 2) | Stage C (Best: Ep 2)
-- | -- | -- | --
Val Accuracy | 99.47% | 99.34% | 99.21%
Val Macro-F1 | 0.992 | 0.990 | 0.988
Val Loss | 0.0218 | 0.0184 | 0.0238

**Confusion Matrix**

<img width="250" height="250" alt="Image" src="https://github.com/user-attachments/assets/4b728b54-b8f6-4170-a314-496d2e5845ed" />

**The experiment have provided exceptional result, next we are saving the Stage B weights to serve as the initial expert layer for the next phase of hierarchical script identification.**

## Reviewer
- [x] @kaldan007

**Comment by eroux** [comment]
there's a bit of work done on this in Vienna by Marco Peer, Channa Li, etc. but unfortunately it seems it was never published. Here are a couple of slides from their 2023 presentation in Vienna

![Image](https://github.com/user-attachments/assets/70b599a9-47df-4328-ac32-20154d14505c)
![Image](https://github.com/user-attachments/assets/97b6c823-17bc-4a1e-98f9-ad155f003a11)
![Image](https://github.com/user-attachments/assets/d6179f1f-efa3-4eb8-a6e4-1c36c5a9700c)

so SIFT is a good keyword to research

also I think it could make sense to just pretrain a visual transformer on all the woodblocks and manuscripts, or maybe a subsample of 100 images per volume, so it "understands" the BDRC images before it tries to classify them

**Comment by webuddhist-bot1** [comment]
## 📋 Card Review

**Card Type:** Developer
**Status:** Ready

### Good Points
- Description is clear and actionable
- Scope/Subtasks well-defined
- Completion Criteria/Definition of Done present

**Comment by karmatai** [comment]
There is already a script classification card here #153. So I am repurposing the card for Retraining DINO V3 on dual grouped class.

**Comment by kaldan007** [comment]
@karmatai I thing dual group is too generalist. I propose going with uchen, ume sub cat 1, ume sub Cat 2, ... remove the multi script and non tibetan as per @eroux suggestion in the channel.

**Comment by karmatai** [comment]
> [@karmatai](https://github.com/karmatai) I thing dual group is too generalist. I propose going with uchen, ume sub cat 1, ume sub Cat 2, ... remove the multi script and non tibetan as per [@eroux](https://github.com/eroux) suggestion in the channel.

@kaldan007 Elie also suggested building on uchen/ume and reuse the weight to train for five classes [ here ](https://discord.com/channels/900268123972845598/1488840320652283944/1501843346073714710)

**Comment by kaldan007** [comment]
cool then lets start with uchen ume

**Comment by gangagyatso4364** [comment]
upload models and datasets along with reviewed benchmark in openpecha huggingface

**Comment by karmatai** [comment]
https://huggingface.co/openpecha/uchen-ume-classifier
